### DISCUSSION TITLE
Add the ability to specify an http adapter for network

### DISCUSSION THREAD

**Discussion by fuz12** [nylo-core/nylo#181] [General]
After creating instance Dio there is no way to define httpClientAdapter. Could you please add this option or allow creating own Dio at all.
Our plans are to switch to http2. We want to do this via package dio_http2_adapter.

**Comment by agordn52** [comment]
Hi @fuz12,

I've added a new method in `nylo_framework` v6.2.2

You can update your pubspec.yaml file, this is what the new method looks like.

``` dart
class ApiService extends NyApiService {
  ApiService({BuildContext? buildContext})
      : super(
          buildContext,
          decoders: modelDecoders,
          initDio: (Dio dio) {
            return dio..httpClientAdapter = Http2Adapter(
              ConnectionManager(idleTimeout: Duration(seconds: 10)),
            );
          },
```

Hope that helps
